docs/reference/files/metadata-yaml-file.rst

```diff
@@ -0,0 +1,896 @@
+    * - ``k8s-api``
+      - The charm deploys if and only if the underlying substrate for the model
+        is Kubernetes.
+      - Since 2.9.23
+
+The Boolean expressions are defined as below:
+
+.. list-table::
+
+    * - ``any-of``
+      - The sub-expression is satisfied if any of the provided child expressions is
+        satisfied.
+    * - ``all-of``
+      - The sub-expression is satisfied if all of the provided child expressions are satisfied.
+
+Examples:
+
+.. dropdown:: Simple example
+
+    .. code-block:: yaml
+
+        assumes:
+        - juju >= 2.9.23
+        - k8s-api
+
+.. dropdown:: Example with a nested expression
+
+    .. code-block:: yaml
+
+        assumes:
+            - any_of:
+                - juju >= 2.9
+                - k8s-api
+
+
+``charm-user``
+--------------
+
+.. important::
+
+    ``charm-user`` was added in Juju 3.6.0. Currently is only supported on Kubernetes
+    charms and has no effect on machine charms.
+
+**Status:** Optional. Recommended for Kubernetes charms.
+
+**Purpose:** The ``charm-user`` key  allows charm authors to specify that their charm hook
+code does not need to be run as root. This key, in addition to ``uid`` + ``uid`` fields in
+``containers``, allows charms to be run rootless. The value of ``root`` ensures the charm
+runs as root. Both ``sudoer`` and ``non-root`` will run as a user other than root. In the
+case of the value ``sudoer``, the charm will be run as a user with access to sudo to
+elevate it's privileges.
+
+**Structure:** The key consists of a single value. One of ``root``, ``sudoer`` or
+``non-root``.
+
+.. code-block:: yaml
+
+    # (Optional) What kind of user is required to run the charm code.
+    # It can be one of root, sudoer or non-root.
+    # Added in Juju 3.6.0. If not specified, root is assumed.
+    charm-user: <one of root, sudoer or non-root>
+
+
+``containers``
+--------------
+
+    See also: `resources <#resources>`_
+
+**Status:** Required for Kubernetes charms (except for proxy charms running on
+Kubernetes).
+
+**Purpose:** The ``containers`` key allows you to define a map of containers to be created
+adjacent to the charm (as a sidecar, in the same pod).
+
+**Structure:** This key consists of a list of containers along with their specification.
+Each container can be specified in terms of ``resource``, ``bases``, ``uid``, ``gid`` and
+``mounts``, where one of either the ``resource`` or the ``bases`` subkeys must be defined, and
+``mounts`` is optional. ``resource`` stands for the OCI image resource used to create the
+container; to use it, specify  an OCI image resource name (that you will then define
+further in the `resources`_ block). ``bases`` is a list of bases to be used
+for resolving a container image, in descending order of preference; to use it, specify a
+base name (for example, ``ubuntu``, ``centos``, ``windows``, ``osx``, ``opensuse``), a
+[channel](https://snapcraft.io/docs/channels), and an architecture. ``mounts`` is a list
```

Comment:
[Hey! You got your Markdown in my rST!](https://www.youtube.com/watch?v=O7oD_oX-Gio)